src/nectarchain/calibration/container/charge.py

```diff
@@ -124,140 +126,212 @@ def __init__(self,charge_hg,charge_lg,peak_hg,peak_lg,run_number,pixels_id,neven
+        # table.write(Path(path)/f"charge_lg_run{self.run_number}.ecsv",format='ascii.ecsv',overwrite=kwargs.get('overwrite',False))
         #
-        #table = Table(self.peak_hg)
-        #table.meta["pixels_id"] = self._pixels_id
-        #table.write(Path(path)/f"peak_hg_run{self.run_number}.ecsv",format='ascii.ecsv',overwrite=kwargs.get('overwrite',False))
+        # table = Table(self.peak_hg)
+        # table.meta["pixels_id"] = self._pixels_id
+        # table.write(Path(path)/f"peak_hg_run{self.run_number}.ecsv",format='ascii.ecsv',overwrite=kwargs.get('overwrite',False))
         #
-        #table = Table(self.peak_lg)
-        #table.meta["pixels_id"] = self._pixels_id
-        #table.write(Path(path)/f"peak_lg_run{self.run_number}.ecsv",format='ascii.ecsv',overwrite=kwargs.get('overwrite',False))
+        # table = Table(self.peak_lg)
+        # table.meta["pixels_id"] = self._pixels_id
+        # table.write(Path(path)/f"peak_lg_run{self.run_number}.ecsv",format='ascii.ecsv',overwrite=kwargs.get('overwrite',False))
 
         hdr = fits.Header()
-        hdr['RUN'] = self._run_number
-        hdr['NEVENTS'] = self.nevents
-        hdr['NPIXELS'] = self.npixels
-        hdr['COMMENT'] = f"The charge containeur for run {self._run_number} with {self._method} method : primary is the pixels id, then you can find HG charge, LG charge, HG peak and LG peak, 2 last HDU are composed of event properties and trigger patern"
-
-        primary_hdu = fits.PrimaryHDU(self.pixels_id,header=hdr)
-        charge_hg_hdu = fits.ImageHDU(self.charge_hg,name = "HG charge")
-        charge_lg_hdu = fits.ImageHDU(self.charge_lg,name = "LG charge")
-        peak_hg_hdu = fits.ImageHDU(self.peak_hg, name = 'HG peak time')
-        peak_lg_hdu = fits.ImageHDU(self.peak_lg, name = 'LG peak time')
-
-        col1 = fits.Column(array = self.event_id, name = "event_id", format = '1I')
-        col2 = fits.Column(array = self.event_type, name = "event_type", format = '1I')
-        col3 = fits.Column(array = self.ucts_timestamp, name = "ucts_timestamp", format = '1K')
-        col4 = fits.Column(array = self.ucts_busy_counter, name = "ucts_busy_counter", format = '1I')
-        col5 = fits.Column(array = self.ucts_event_counter, name = "ucts_event_counter", format = '1I')
-        col6 = fits.Column(array = self.multiplicity, name = "multiplicity", format = '1I')
+        hdr["RUN"] = self._run_number
+        hdr["NEVENTS"] = self.nevents
+        hdr["NPIXELS"] = self.npixels
+        hdr["COMMENT"] = (
+            f"The charge containeur for run {self._run_number} with {self._method} "
+            f"method : primary is the pixels id, then you can find HG charge, "
+            f"LG charge, HG peak and LG peak, 2 last HDU are composed of event "
+            f"properties and trigger patern"
+        )
+
+        primary_hdu = fits.PrimaryHDU(self.pixels_id, header=hdr)
+        charge_hg_hdu = fits.ImageHDU(self.charge_hg[start:end], name="HG charge")
```

Comment:
For instance here, the default behavior should allow for the full `charge_hg` attribute to be stored and written to disk.